tests/integration/ha_tests/test_self_healing.py

```diff
@@ -543,3 +558,119 @@ async def test_network_cut_without_ip_change(
+    app = await app_name(ops_test)
+    dbname = f"{APPLICATION_NAME.replace('-', '_')}_first_database"
+    connection_string, _ = await get_db_connection(ops_test, dbname=dbname)
+
+    # Start an application that continuously writes data to the database.
+    await start_continuous_writes(ops_test, app)
+
+    logger.info("checking whether writes are increasing")
+    await are_writes_increasing(ops_test)
+
+    # Connect to the database.
+    # Create test data.
+    logger.info("connect to DB and create test table")
+    await create_test_data(connection_string)
+
+    # Test to check the use of different versions postgresql.
+    # Release of a new version of charm with another version of postgresql,
+    # it is necessary to implement a test that will check the use of different versions of postgresql.
+
+    unit_ip_addresses = []
+    primary_storage = ""
+    for unit in ops_test.model.applications[app].units:
+        # Save IP addresses of units
+        unit_ip_addresses.append(await get_unit_ip(ops_test, unit.name))
+
+        # Save detached storage ID
+        if await unit.is_leader_from_status():
+            primary_storage = storage_id(ops_test, unit.name)
+
+    logger.info(f"get storage id app: {SECOND_APPLICATION}")
+    second_storage = ""
+    for unit in ops_test.model.applications[SECOND_APPLICATION].units:
+        if await unit.is_leader_from_status():
+            second_storage = storage_id(ops_test, unit.name)
+            break
+
+    # Scale the database to zero units.
+    logger.info("scaling database to zero units")
+    await scale_application(ops_test, app, 0)
+    await scale_application(ops_test, SECOND_APPLICATION, 0)
+
+    # Checking shutdown units.
+    for unit_ip in unit_ip_addresses:
+        try:
+            resp = requests.get(f"http://{unit_ip}:8008")
+            assert (
+                resp.status_code != 200
+            ), f"status code = {resp.status_code}, message = {resp.text}"
+        except requests.exceptions.ConnectionError:
+            assert True, f"unit host = http://{unit_ip}:8008, all units shutdown"
+        except Exception as e:
+            assert False, f"{e} unit host = http://{unit_ip}:8008, something went wrong"
+
+    # Scale up to one unit.
+    logger.info("scaling database to one unit")
+    await add_unit_with_storage(ops_test, app=app, storage=primary_storage)
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME, APPLICATION_NAME], status="active", timeout=1500
+    )
+
+    connection_string, _ = await get_db_connection(ops_test, dbname=dbname)
+    logger.info("checking whether writes are increasing")
+    await are_writes_increasing(ops_test)
+
+    logger.info("check test database data")
+    await validate_test_data(connection_string)
+
+    logger.info("database scaling up to two units using third-party cluster storage")
+    new_unit = await add_unit_with_storage(
+        ops_test, app=app, storage=second_storage, is_blocked=True
+    )
```

Comment:
nit: IMHO, it worth to check: we are blocked with the right message (foreign disk).